src/app/pages/PerpetualPage/components/TradeDialog/types.tsx

```diff
@@ -0,0 +1,67 @@
+import { TradingPosition } from '../../../../../types/trading-position';
+import { Transaction } from '../../../../../store/global/transactions-store/types';
+import { PerpetualPageModals, PerpetualTrade } from '../../types';
+import { PerpetualPair } from '../../../../../utils/models/perpetual-pair';
+import { Nullable } from '../../../../../types';
+
+export enum TradeDialogStep {
+  review = 'review',
+  approval = 'approval',
+  processing = 'processing',
+}
+
+export type TradeAnalysis = {
+  amountChange: number;
+  amountTarget: number;
+  marginChange: number;
+  marginTarget: number;
+  roe: number;
+  leverageTarget: number;
+  entryPrice: number;
+  liquidationPrice: number;
+  tradingFee: number;
+};
+
+export type PerpetualTxTrade = {
+  method: 'trade';
```

Comment:
I would extract methods to an enum (e.g. `PerpetualTxMethods`).

Reply:
Yep, you're right.